crates/trie/parallel/src/parallel_proof.rs

```diff
@@ -0,0 +1,252 @@
+            walker,
+            hashed_cursor_factory.hashed_account_cursor().map_err(ProviderError::Database)?,
+        );
+        while let Some(account_node) =
+            account_node_iter.try_next().map_err(ProviderError::Database)?
+        {
+            match account_node {
+                TrieElement::Branch(node) => {
+                    hash_builder.add_branch(node.key, node.value, node.children_are_in_trie);
+                }
+                TrieElement::Leaf(hashed_address, account) => {
+                    let storage_multiproof = match storage_proofs.remove(&hashed_address) {
+                        Some(rx) => rx.recv().map_err(|_| {
+                            ParallelStateRootError::StorageRoot(StorageRootError::Database(
+                                reth_db::DatabaseError::Other(format!(
+                                    "channel closed for {hashed_address}"
+                                )),
+                            ))
+                        })??,
+                        // Since we do not store all intermediate nodes in the database, there might
+                        // be a possibility of re-adding a non-modified leaf to the hash builder.
+                        None => {
+                            tracker.inc_missed_leaves();
+                            storage_multiproof(
+                                trie_cursor_factory.clone(),
+                                hashed_cursor_factory.clone(),
+                                Default::default(), // TODO: check?
+                                hashed_address,
+                                targets.get(&hashed_address).map_or(Vec::new(), |slots| {
+                                    slots.iter().map(Nibbles::unpack).collect()
+                                }),
+                            )
+                            .map_err(ProviderError::Database)?
+                        }
+                    };
+
+                    // Encode account
+                    account_rlp.clear();
+                    let account = TrieAccount::from((account, storage_multiproof.root));
+                    account.encode(&mut account_rlp as &mut dyn BufMut);
+
+                    hash_builder.add_leaf(Nibbles::unpack(hashed_address), &account_rlp);
+                    storages.insert(hashed_address, storage_multiproof);
+                }
+            }
+        }
+        let _ = hash_builder.root();
+
+        #[cfg(feature = "metrics")]
+        self.metrics.record_state_trie(tracker.finish());
+
+        Ok(MultiProof { account_subtree: hash_builder.take_proof_nodes(), storages })
+    }
+}
+
+/// Generate a storage multiproof according to specified targets.
+pub fn storage_multiproof<T, H>(
+    trie_cursor_factory: T,
+    hashed_cursor_factory: H,
+    prefix_set: PrefixSet, // must already include targets
+    hashed_address: B256,
+    targets: Vec<Nibbles>,
+) -> Result<StorageMultiProof, DatabaseError>
```

Comment:
we have `StorageProof::storage_multiproof` for this now

Reply:
thx! done here https://github.com/paradigmxyz/reth/pull/12403/commits/935df37a857a630ce9d919dba83816b5e364f233